File: backend/src/main/kotlin/ru/itmo/db/coursework/api/dto/MachineRecipeDto.kt
package ru.itmo.db.coursework.api.dto

import com.fasterxml.jackson.annotation.JsonProperty

data class MachineRecipeDto (
        var id: Int,
        @get:JsonProperty("circuit_board_id")
        var circuitBoardId: Int,
        @get:JsonProperty("circuit_board_amount")
        var circuitBoardAmount: Int,
        @get:JsonProperty("work_time")
        var workTime: Long,
        var price: Long,
        var teas: Map<Int, Int>
)

File: frontend/src/api/dto/MachineRecipeDto.ts
export default class MachineRecipeDto {
  constructor(
    public id: number,
    public circuit_board_id: number,
    public circuit_board_amount: number,
    public work_time: number,
    public price: number,
    public teas: { [key: number]: number }
  ) {}
}